unifiednlp-base/src/main/java/org/microg/tools/selfcheck/NlpStatusChecks.java

```diff
@@ -33,53 +34,102 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 
 import static android.content.Context.LOCATION_SERVICE;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.location.Address;
 import static android.location.LocationManager.NETWORK_PROVIDER;
+import android.os.IBinder;
 import static org.microg.nlp.api.Constants.LOCATION_EXTRA_BACKEND_COMPONENT;
 import static org.microg.tools.selfcheck.SelfCheckGroup.Result.Negative;
 import static org.microg.tools.selfcheck.SelfCheckGroup.Result.Positive;
 import static org.microg.tools.selfcheck.SelfCheckGroup.Result.Unknown;
 
-public class NlpStatusChecks implements SelfCheckGroup {
-    private Location mLastLocation;
+public class NlpStatusChecks extends Service implements SelfCheckGroup {
```

Comment:
OK, I can do that. I know how to create specific service, but I don't know how to implement callback from some service actually. Could you give me some help or example?